packages/fast-usdc/src/util/noble.js

```diff
@@ -0,0 +1,98 @@
+import { DirectSecp256k1HdWallet, Registry } from '@cosmjs/proto-signing';
+import { AminoTypes, SigningStargateClient } from '@cosmjs/stargate';
+import { nobleAminoConverters, nobleProtoRegistry } from '@nick134-bit/noblejs';
+import axios from 'axios';
+
+const createMsgRegisterAccount = (
+  /** @type {string} */ signer,
+  /** @type {string} */ recipient,
+  /** @type {string} */ channel,
+) => {
+  return {
+    typeUrl: '/noble.forwarding.v1.MsgRegisterAccount',
+    value: {
+      signer,
+      recipient,
+      channel,
+    },
+  };
+};
+
+export const registerFwdAccount = async (
+  /** @type {string} */ nobleSeed,
+  /** @type {string} */ nobleToAgoricChannel,
+  /** @type {string} */ nobleRpc,
+  /** @type {string} */ recipient,
+  out = console,
+) => {
+  out.log('registering fwd account on noble');
+  const wallet = await DirectSecp256k1HdWallet.fromMnemonic(nobleSeed, {
+    prefix: 'noble',
+  });
+  out.log('got noble wallet from seed');
+  const accounts = await wallet.getAccounts();
+  const signerAddress = accounts[0].address;
+  out.log(`got signer ${signerAddress}`);
+  const msg = createMsgRegisterAccount(
+    signerAddress,
+    recipient,
+    nobleToAgoricChannel,
+  );
+  const fee = {
+    amount: [
+      {
+        denom: 'uusdc',
+        amount: '20000',
+      },
+    ],
+    gas: '200000',
+  };
+  out.log('signing message', msg);
+  const clientWithSigner = await SigningStargateClient.connectWithSigner(
+    nobleRpc,
```

Comment:
`nobleRpc` should be an RPC client object rather than a string. Something like this should work, based on [some endo playground experience](https://github.com/endojs/playground/blob/main/packages/ag-trade/src/smartWallet.js#L200-L217):

Use [makeTendermintRpcClient(url, fetch)](https://github.com/Agoric/agoric-sdk/blob/d19b293d5b28a2d78754eb7912771bdb66d5fb37/packages/casting/src/makeHttpClient.js#L28C14-L28C37) from casting and the rest comes from cosmjs:

```js
import { Tendermint34Client } from '@cosmjs/tendermint-rpc';

...

  const cometClient = await Tendermint34Client.create(rpcClient);
  const clientWithSigner = await SigningStargateClient.createWithSigner(
    cometClient,
    ...
```

Reply:
Why though? It adds more dependencies and line of code, while the `SigningStargateClient` interface supports a string

Reply:
> It adds more dependencies

It adds lines in `package.json` maybe, but the dependencies are already there, indirectly.

> `SigningStargateClient` interface supports a string

yes, but that string separates designation from authority. If we pass a string, `SigningStargateClient` is going to use ambient access to the network - `fetch` or the like. How would we test `registerFwdAccount`? Perhaps there are tools that use global mutable state to override `fetch`. That sort of spooky action-at-a-distance is difficult to manage at scale. And it means `registerFwdAccount` doesn't support patterns of reuse such as passing in an RPCClient with built-in rate-limiting.